README.rst

```diff
@@ -1,102 +1,208 @@
 Klister
 ------------
 
-This repository contains a partial implementation of a macro expander
-in which macros can get *stuck*. The idea is that if a macro depends
-on information that will be the result of later macro expansion or
-type checking, it can pause its execution until that information has
-become available.
-
-For a description of one way in which stuck macros might be useful,
-please see this `talk`_.
-
-.. _talk: https://www.youtube.com/watch?v=nUvKoG_V_U0
-
-
-Installation
-============
-
-Klister is written in Haskell, and is built with HPack and Cabal. You can build
-it like so::
-
-    hpack package.yaml
-    cabal v2-build
-
-You can use ``v2-install`` instead of ``v2-build`` to install the Klister
-executable and associated files to your home directory.
-
-Usage
-=====
-
-Run ``klister repl`` to be dropped into a read-eval-print loop where you can try
-out writing some Klister expressions.
-
-``klister run file.kl`` will evaluate the examples and run the IO
-actions in ``file.kl``.
+Klister [`TyDe 2020`_, `video`_] is a research-stage programming language which combines
+the core features of Racket and Haskell into a single language. It is named
```

Comment:
I'd talk about the ML family of languages here, rather than Haskell per se. We're much closer to old-school ML in the type system here, though we do use monads to sequence effects.

Reply:
(e.g. we don't support polymorphic recursion)

Reply:
I really do want to mention Haskell early and prominently, because Haskell users are a key demographic for this project:

1. Klister is implemented in Haskell, so Haskellers are our most likely contributors
2. We're both involved in the Haskell community and thus are going to be sending a disproportionate amount of Haskellers to this README compared to visitors from other communities
3. The language to which Klister is the most similar is Hackett, which does position itself as a combination of Racket and Haskell.

Thus, I would prefer to mention Haskell early, and then to enumerate the differences which makes Klister closer to ML than to Haskell, than to start with ML and then enumerate the ways in which Klister inches towards Haskell.

What if I mentioned all three of Racket, ML, and Haskell in the first paragraph, would that be a good compromise?

Reply:
I tried my counter-suggestion in 9232b80a847b8c3b51fceeca5bca6c6ab48eeabc, let me know what you think!